server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java

```diff
@@ -139,6 +142,10 @@ static Mapping group(String mapping) {
         return of(Type.Group, mapping);
     }
 
+    static Mapping alias(String mapping) {
+        return of(Type.Alias, mapping);
+    }
+
     static Type detectType(String input) {
```

Comment:
This method is quite crazy and maybe this is time to write it in an automatic fashion to avoid modifying it every time we do add a mapping type.

How about:

```
Type.values().stream()
     .filter(type -> !type.asPrefix().isEmpty())
     .filter(type -> input.startsWith(type.asPrefix())
     .findAny()
     .orElse(Type.Address);
```

This would make the job of next implementers way easier....